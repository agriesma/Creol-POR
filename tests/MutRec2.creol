
interface Test
begin
  with Test op m
end

class Test
contracts Test
begin
  op init == var a: Test; a := new Test; a.m(;)
  op run == skip
  with Test
    op m == caller.m(;)
end
