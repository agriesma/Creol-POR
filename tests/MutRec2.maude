load creol-interpreter
mod PROGRAM is
pr CREOL-INTERPRETER .
op init : -> Configuration [ctor] .
eq init =
< "Test" : Cl | Inh: noInh, Par: noVid, Att: noSubst, Mtds: 
  < "init" : Mtdname | Param: noVid, Latt: "a" |-> null, Code: "a" ::= new "A" ( emp ) ; "sync.0" ! "a" . "m" ( emp ) ; ( "sync.0" ? ( noVid ) )  ; return ( emp ) > *
  < "run" : Mtdname | Param: noVid, Latt: noSubst, Code: skip ; return ( emp ) > *
  < "m" : Mtdname | Param: noVid, Latt: noSubst, Code: "sync.0" ! "caller" . "m" ( emp ) ; ( "sync.0" ? ( noVid ) )  ; return ( emp ) >, Ocnt: 0 >

.
endm
