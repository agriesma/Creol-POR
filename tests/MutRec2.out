==========================================
reduce in PROGRAM : init .
result Class: 
< "Test" : Cl | Inh: noInh, Par: noVid, Att: noSubst, Mtds:
  < "init" : Mtdname | Param: noVid, Latt: "a" |-> null, Code: "a" ::= new
    "Test"(emp) ; "sync.0" ! "a" . "m"(emp) ; "sync.0" ?(noVid) ; return(emp) >
    *
  < "m" : Mtdname | Param: noVid, Latt: noSubst, Code: "sync.0" ! "caller" .
    "m"(emp) ; "sync.0" ?(noVid) ; return(emp) > *
  < "run" : Mtdname | Param: noVid, Latt: noSubst, Code: skip ; return(emp) >,
  Ocnt: 0 >
Bye.
